File: eval/fixtures/cross-repo/acme-core/src/store.ts
import { createTask, overdue, type Task } from './task.ts';

export class MemStore {
  private tasks = new Map<string, Task>();

  add(id: string, title: string, dueAt: number): Task {
    const task = createTask(id, title, dueAt);
    this.tasks.set(id, task);
    return task;
  }

  get(id: string): Task | undefined {
    return this.tasks.get(id);
  }

  overdueTasks(now: number): Task[] {
    return [...this.tasks.values()].filter((t) => overdue(t, now));
  }
}

File: eval/fixtures/cross-repo-py/py-core/taskcore/store.py
"""In-memory task store built on the domain helpers."""
from taskcore.task import Task, create_task, overdue


class MemStore:
    def __init__(self) -> None:
        self._tasks: dict[str, Task] = {}

    def add(self, id: str, title: str, due_at: int) -> Task:
        task = create_task(id, title, due_at)
        self._tasks[id] = task
        return task

    def get(self, id: str) -> Task | None:
        return self._tasks.get(id)

    def overdue_tasks(self, now: int) -> list[Task]:
        return [t for t in self._tasks.values() if overdue(t, now)]
